warehouse/accounts/forms.py

```diff
@@ -562,11 +574,11 @@ def validate_username_or_email(self, field):
         if "@" in field.data:
             # Additional checks for the validity of the address
             try:
-                Address(addr_spec=field.data)
-            except (IndexError, ValueError, HeaderParseError):
+                email_validator.validate_email(field.data, check_deliverability=True)
+            except email_validator.EmailNotValidError as e:
                 raise wtforms.validators.ValidationError(
                     message=INVALID_PASSWORD_MESSAGE
```

Comment:
While we're here, is this the right message?

Reply:
Heh, nope. Thanks for pointing that out, will add a commit to use a better message.

Reply:
Added in b1d6704aeaeb754246fd3ce3de9f3848a8ee4b40